src/app/components/dynamicdialog/dynamicdialog.ts

```diff
@@ -215,12 +216,40 @@ export class DynamicDialogComponent implements AfterViewInit, OnDestroy {
         @SkipSelf() @Optional() private parentDialog: DynamicDialogComponent
     ) {}
 
+    ngOnInit() {
+        this.setPosition(this.config.position);
+    }
+
     ngAfterViewInit() {
         this.loadChildComponent(this.childComponentType!);
         this.ariaLabelledBy = this.getAriaLabelledBy();
         this.cd.detectChanges();
     }
 
+    setPosition(value: string) {
```

Comment:
I would like to suggest this code to avoid `switch-case` and use a map:

```
setPosition(value: string) {
  const transformOptionsMap = {
    'topleft': 'translate3d(-100%, 0px, 0px)',
    'bottomleft': 'translate3d(-100%, 0px, 0px)',
    'left': 'translate3d(-100%, 0px, 0px)',
    'topright': 'translate3d(100%, 0px, 0px)',
    'bottomright': 'translate3d(100%, 0px, 0px)',
    'right': 'translate3d(100%, 0px, 0px)',
    'bottom': 'translate3d(0px, 100%, 0px)',
    'top': 'translate3d(0px, -100%, 0px)',
  };

  this.transformOptions = transformOptionsMap[value] || 'scale(0.7)';
  }
```
  Tested first because I didn't. The code is a little bit shorter and cleaner in my opinion.
  
  Thanks!

Reply:
OK, I will change and test my code and commit again.